pandas/tslib.pyx

```diff
@@ -1052,8 +1056,11 @@ cdef class _Timestamp(datetime):
             other = Timestamp(other)
 
             # validate tz's
-            if get_timezone(self.tzinfo) != get_timezone(other.tzinfo):
-                raise TypeError("Timestamp subtraction must have the same timezones or no timezones")
+            if get_timezone(self.tzinfo) != get_timezone(other.tzinfo): 
```

Comment:
this is not enough, you need to simply do the conversion to 'UTC'. so an already UTC tz will be fine, a non-UTC zone will be converted as well, BUT a non-timezone aware will raise, which should be raised to a higher level.